syntax = "proto3";

package consumer;

service Consumer {
    rpc GetLatestMessage (GetLatestMessageRequest) returns (GetLatestMessageResponse);
    rpc ListenForNewMessages (ListenForNewMessagesRequest) returns (stream ListenForNewMessagesResponse);
}

message GetLatestMessageRequest {}

message GetLatestMessageResponse {
    string data = 1;
}

message ListenForNewMessagesRequest {}

message ListenForNewMessagesResponse {
    string data = 1;
}